### PR TITLE
chore: prepare 0.4.6

### PR DESCRIPTION
**Pull request by Yohe-Am**
- Bump version to 0.4.6-0
- Add sanity tests for published SDKs
- Bump deno to 1.45.2
- Bump rust to 1.79.0
- Fix myriad of bugs

#### Migration notes

...

- [x] The change comes with new or modified tests
- [ ] Hard-to-understand functions have explanatory comments
- [ ] End-user documentation is updated to reflect the change

## Summary by CodeRabbit

- **New Features**
  - Introduced new logging capabilities in the `ConnectedEngine` with adjustable logging levels.
  - Implemented cleanup procedures in tests to enhance resource management.

- **Bug Fixes**
  - Fixed import paths for permissions to ensure correct functionality in tests and applications.

- **Version Updates**
  - Incremented version numbers across multiple projects and packages to reflect ongoing development and improvements.
  
- **Documentation**
  - Added comments to clarify code behavior and potential future considerations in various modules. 

- **Refactor**
  - Optimized string handling in several functions and adjusted method signatures for improved clarity and efficiency.

### CODE REVIEW

**Review by zifeo — APPROVED** [review]

**Comment by Yohe-Am** [comment]
@coderabbitai review

File: dev/tasks-dev.ts [hunk 1]
```diff
@@ -63,7 +63,7 @@ const tasks: Record<string, DenoTaskDefArgs> = {
             "-f",
             file,
           ])
-        } down --remove-orphans`;
+        } down --remove-orphans --volumes`;
```
Comment (zifeo) [inline dev/tasks-dev.ts:66]: fyi: https://github.com/zifeo/substantial/pull/2/files#r1685540051

**Review by zifeo — DISMISSED** [review]

File: dev/tasks-dev.ts (hunk 1 above)
Comment (Yohe-Am) [reply dev/tasks-dev.ts:66]: Nice, thanks. I actually added that in a futile attempt to solve this issue with the bitnami containers and coder. Whenever I restart the coder workspace, the old containers seem to leave "something" around that prevents new ones from working. Specifically, they leave files that are under the wrong owner that prevents the new containers from working.

Why is "something" in scare quotes? Well, these containers don't have any volumes attached. What's more, the conflict happens even after I `compose down` and `compose up` again. Even with the `--volumes` flag. It's as if the images got corrupted but I'm not sure. 

The only solution I've found is to `docker system prune -a` and clean out everything.

**Review by zifeo — DISMISSED** [review]

**Review by afmika — APPROVED** [review]

**Review by Natoandro — APPROVED** [review]

File: typegate/tests/e2e/published/published_test.ts
[the remark this answers is not in the document]
Comment (Yohe-Am) [reply typegate/tests/e2e/published/published_test.ts:1]: The other way around you mean? Most of the tests run on the current commit. My hope for this test is that we'll see the issues right away in the PR that's opened by the release pipeline to bump to the pre-release version.

File: typegate/tests/metagen/typegraphs/identities/py/handlers_types.py [hunk 2]
```diff
@@ -116,7 +116,7 @@ class CompositesArgs(Struct):
 @dataclass
 class Composites(Struct):
     opt: Union[str, None]
-    either: Union["Branch2", "Primitives"]
+    either: Union["Primitives", "Branch2"]
```
[the remark this answers is not in the document]
Comment (Yohe-Am) [reply typegate/tests/metagen/typegraphs/identities/py/handlers_types.py:119]: Yep, I believe switching out the `HashMap`s in the mdk_python impl for `BTreeMap`s should do the trick. Will take care of it in my python client PR.

[the remark this answers is not in the document]
Comment (afmika) [reply typegate/tests/metagen/typegraphs/identities/py/handlers_types.py:119]: `IndexMap` will suffice imo if we only care about the insertion order.

**Review by Natoandro — APPROVED** [review]

**Review by zifeo — DISMISSED** [review]

**Review by Natoandro — APPROVED** [review]

**Review by Natoandro — APPROVED** [review]

**Pull request merged**